ggshield/core/config/config.py

```diff
@@ -86,13 +86,22 @@ def get_instance_name_and_source(self) -> Tuple[str, ConfigSource]:
         The instance name (defaulting to URL) of the selected instance
         priority order is:
         - set from the command line (by setting cmdline_instance_name)
+          - in case the user set the api url instead of dashboard url, we replace it
         - GITGUARDIAN_INSTANCE env var
         - GITGUARDIAN_API_URL env var
         - in local user config (in user_config.dashboard_url)
         - in global user config (in user_config.dashboard_url)
         - the default instance
         """
         if self._cmdline_instance_name:
+            if (
+                "api" in self._cmdline_instance_name
+                and "gitguardian" in self._cmdline_instance_name
+            ):
```

Comment:
This looks a bit too generic to me. We could run into a corner case where an on-prem URL contains both `api` and `gitguardian`. I'd rather use a regex matching on `^https://api(\.[a-z0-9]+)?\.gitguardian\.com`.